src/app/components/product/product.component.spec.ts

```diff
@@ -36,28 +35,35 @@ describe('AppComponent => ProductComponent', () => {
       })
       .compileComponents();
 
-    fixture = TestBed.createComponent(ProductComponent);
+    fixture = TestBed.createComponent(TestComponent);
+    loader = TestbedHarnessEnvironment.loader(fixture);
     component = fixture.componentInstance;
 
     component.product = product;
     fixture.detectChanges();
+    await fixture.whenStable();
   });
 
   it('should be created', () => {
```

Comment:
I suggest merging all 4 cases here into one called **_it('renders component')_**

Since they don't have a strong logical value to be split like that.